contrib/opentelemetry/go.mod

```diff
@@ -1,12 +1,14 @@
 module go.temporal.io/sdk/contrib/opentelemetry
 
-go 1.20
+go 1.21
+
+toolchain go1.22.2
```

Comment:
Reading https://go.dev/ref/mod#go-mod-file-toolchain, I am trusting this has no effect on users using 1.21?

Reply:
It has no effect on users, that  being said it messes with our CI because it causes the unit tests to always be run with  GO version `1.22.2` so I will remove.